vm/src/bench.rs

```diff
@@ -151,3 +170,402 @@ fn trace_as_trace(fvm_trace: fvm::trace::ExecutionTrace) -> ExecutionTrace {
     }
     ExecutionTrace::new(events)
 }
+#[derive(Debug, Clone)]
+pub struct TestData {
+    circ_supply: TokenAmount,
+    price_list: PriceList,
+}
+
+pub type Result<T> = std::result::Result<T, ExecutionError>;
+pub struct BenchKernel<B: Blockstore + 'static> {
+    kernel: DefaultKernel<DefaultCallManager<DefaultMachine<B, FakeExterns>>>,
+    test_data: TestData,
+}
+
+impl<B> Kernel for BenchKernel<B>
+where
+    B: Blockstore + 'static,
+{
+    type CallManager = DefaultCallManager<DefaultMachine<B, FakeExterns>>;
+
+    fn into_inner(self) -> (Self::CallManager, fvm::kernel::BlockRegistry)
+    where
+        Self: Sized,
+    {
+        self.kernel.into_inner()
+    }
+
+    fn new(
+        mgr: Self::CallManager,
+        blocks: fvm::kernel::BlockRegistry,
+        caller: ActorID,
+        actor_id: ActorID,
+        method: fvm_shared::MethodNum,
+        value_received: fvm_shared::econ::TokenAmount,
+        read_only: bool,
+    ) -> Self
+    where
+        Self: Sized,
+    {
+        let kernel =
+            DefaultKernel::new(mgr, blocks, caller, actor_id, method, value_received, read_only);
+
+        let price_list = kernel.price_list().clone();
+
+        println!("Making a new kernel");
+
+        Self { kernel, test_data: TestData { circ_supply: TOTAL_FILECOIN.clone(), price_list } }
+    }
+
+    fn machine(&self) -> &<Self::CallManager as fvm::call_manager::CallManager>::Machine {
+        self.kernel.machine()
+    }
+}
+
+impl<B> ActorOps for BenchKernel<B>
+where
+    B: Blockstore,
+{
+    fn resolve_address(&self, address: &Address) -> Result<ActorID> {
+        self.kernel.resolve_address(address)
+    }
+
+    fn get_actor_code_cid(&self, id: ActorID) -> Result<Cid> {
+        self.kernel.get_actor_code_cid(id)
+    }
+
+    fn next_actor_address(&self) -> Result<Address> {
+        self.kernel.next_actor_address()
+    }
+
+    fn create_actor(
+        &mut self,
+        code_id: Cid,
+        actor_id: ActorID,
+        delegated_address: Option<Address>,
+    ) -> Result<()> {
+        self.kernel.create_actor(code_id, actor_id, delegated_address)
+    }
+
+    fn get_builtin_actor_type(&self, code_cid: &Cid) -> Result<u32> {
+        self.kernel.get_builtin_actor_type(code_cid)
+    }
+
+    fn get_code_cid_for_type(&self, typ: u32) -> Result<Cid> {
+        self.kernel.get_code_cid_for_type(typ)
+    }
+
+    #[cfg(feature = "m2-native")]
+    fn install_actor(&mut self, _code_id: Cid) -> Result<()> {
+        Ok(())
+    }
```

Comment:
Can we omit this method since we don't set this feature? Else should probably fail rather than succeed.